### PR TITLE
Correct Sign/VerifyRecover attribute checking

### PR DESCRIPTION
**Pull request by ifranzki**
C_SignRecoverInit and C_VerifyRecoverInit must check for key attributes CKA_SIGN_RECOVER and CKA_VERIFY_RECOVER instead of CKA_SIGN and CKA_VERIFY being TRUE, otherwise the operations must fail with CKR_KEY_FUNCTION_NOT_PERMITTED.

### CODE REVIEW

File: usr/lib/pkcs11/common/sign_mgr.c [hunk 1]
```diff
@@ -60,21 +60,36 @@ CK_RV sign_mgr_init(STDLL_TokData_t *tokdata,
...
+                                     &attr);
+        if (rc == FALSE) {
+            TRACE_ERROR("Could not find CKA_SIGN_RECOVER for the key.\n");
+            return CKR_KEY_FUNCTION_NOT_PERMITTED;
+        }
+        flag = *(CK_BBOOL *) attr->pValue;
+        if (flag != TRUE) {
+            TRACE_ERROR("%s\n", ock_err(ERR_KEY_FUNCTION_NOT_PERMITTED));
+            return CKR_KEY_FUNCTION_NOT_PERMITTED;
+        }
```
Comment (p-steuer) [inline usr/lib/pkcs11/common/sign_mgr.c]: Code for setting and checking flag is doubled in if-else. Could you move it below if-else ?

File: usr/lib/pkcs11/common/verify_mgr.c [hunk 2]
```diff
@@ -61,21 +61,36 @@ CK_RV verify_mgr_init(STDLL_TokData_t *tokdata,
...
+                                     &attr);
+        if (rc == FALSE) {
+            TRACE_ERROR("Could not find CKA_VERIFY_RECOVER for the key.\n");
+            return CKR_KEY_FUNCTION_NOT_PERMITTED;
+        }
+        flag = *(CK_BBOOL *) attr->pValue;
+        if (flag != TRUE) {
+            TRACE_ERROR("%s\n", ock_err(ERR_KEY_FUNCTION_NOT_PERMITTED));
+            return CKR_KEY_FUNCTION_NOT_PERMITTED;
+        }
```
Comment (p-steuer) [inline usr/lib/pkcs11/common/verify_mgr.c]: Same here.

File: testcases/common/common.c [hunk 3]
```diff
@@ -38,6 +38,20 @@ int mech_supported(CK_SLOT_ID slot_id, CK_ULONG mechanism)
     return (rc == CKR_OK);
 }
 
+int mech_supported_flags(CK_SLOT_ID slot_id, CK_ULONG mechanism, CK_FLAGS flags)
+{
+    CK_MECHANISM_INFO mech_info;
+    int rc;
```
Comment (p-steuer) [inline testcases/common/common.c]: Please add a newline below declarations.

**Comment by p-steuer** [comment]
Done with review. Looks good.

File: usr/lib/pkcs11/common/sign_mgr.c (hunk 1 above)
Comment (ifranzki) [reply usr/lib/pkcs11/common/sign_mgr.c]: fixed

File: usr/lib/pkcs11/common/verify_mgr.c (hunk 2 above)
Comment (ifranzki) [reply usr/lib/pkcs11/common/verify_mgr.c]: fixed

File: testcases/common/common.c (hunk 3 above)
Comment (ifranzki) [reply testcases/common/common.c]: fixed

**Pull request merged**